File: 8KYU/I love you, a little , a lot, passionately ... not at all/solution.cs
using System;

public class Kata
{
  public static string HowMuchILoveYou(int nb_petals)
  {
    var phrases = new[] {"I love you", "a little", "a lot", "passionately", "madly", "not at all"};
  var petal = 0;  
  for(int i = 1; i < nb_petals; i++)
  {
    petal++;
    if(petal == 6)
      petal = 0;
  }
  
  return phrases[petal];
  }
}

File: 8KYU/I love you, a little , a lot, passionately ... not at all/solution.c
#include <stddef.h> // NULL

const char* how_much_i_love_you(int nb_petals) {
  const char *phrases[6];
  phrases[0] = "I love you";
  phrases[1] = "a little";
  phrases[2] = "a lot";
  phrases[3] = "passionately";
  phrases[4] = "madly";
  phrases[5] = "not at all";
  
  return phrases[(nb_petals -1) % 6];
}